### PR TITLE
Adds cli and test to extract a cube from a cubelist

### PR DESCRIPTION
**Pull request by mspelman07**
EPP_ticket: https://metoffice.atlassian.net/browse/EPPT-1799
acceptance_test_data: https://github.com/metoppv/improver_test_data/pull/59

Adds in the ability to extract a cube from a cubelist based on the provided name

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by cpelley — CHANGES_REQUESTED** [review]
What's the motivation around needing this?  I don't see anything from the Jira issue.

I'm not so sure about an explicit extract processing step (plugin) like this.  Without any further information, feels like this is substituting for what should be the responsibility of each processing plugin to utilise relevant cube metadata.

As I say, perhaps with additional context??

Cheers

**Comment by mspelman07** [comment]
@cpelley We need this for the relative vorticity workflow. In that workflow we calculate the u and v components of the wind using the resolve-wind-components improver CLI which returns a cubelist of the u and v components. We then need to use the gradient-between-adjacent-grid-squares cli to calculate dv/dx and du/dy. To do this we need to split up the cubelist into individual cubes so we can apply the different processing to each cube. 

As far as I know there is no way to define this directly in a workflow. I thought about creating a new plugin called relative_vorticity which would take in the cubelist and apply the processing as required but this felt more niche and less re-usable than a generic plugin to extract a cube from a cubelist.

I'm not up-to-date on the work you've been doing so it might be there's a better way of doing this sort of thing I'm not aware of.

**Comment by cpelley** [comment]
Thanks for the reply @mspelman07.  Let's have a quick chat offline 👍

File: improver/cli/cubelist_extract.py
Comment (mo-robertneal) [inline improver/cli/cubelist_extract.py:19]:
```suggestion
            A cubelist containing exactly one cube with the provided name to
```

**Review by mo-robertneal — COMMENTED** [review]
The code changes look fine to me and the unit tests all passed. I have suggested one change to one of the comments though. Happy to approve once this very small change is made.

File: improver/cli/cubelist_extract.py
Comment (mspelman07) [reply improver/cli/cubelist_extract.py]: Done

**Review by mo-robertneal — APPROVED** [review]

**Review by Katie-Howard — CHANGES_REQUESTED** [review]
I've looked through the changes and it all seems to work well. The code does what it's supposed to and extracts a cube from a cubelist. The tests also seem sensible, though I've left a comment about one potential change. I've run both the unit and acceptance tests and they all pass.

**Review by Katie-Howard — COMMENTED** [review]
I'm happy this is correct after Carwyn's comment 👍 So I'll approve this ticket.

**Review by Katie-Howard — APPROVED** [review]

**Pull request merged**
